CacheManagerManagementTest test failures

master (commit 94d3d2c713ed0f6c49a2922330cb21e3785be449)

Failed on Sonar : http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.x-sonar/296/#showFailuresLink

```
com.hazelcast.cache.jsr.CacheManagerManagementTest.testJMXGetsCacheAdditionsAndRemovals
com.hazelcast.cache.jsr.CacheManagerManagementTest.testJMXGetsCacheAdditionsAndRemovals
com.hazelcast.cache.jsr.CacheManagerManagementTest.testCacheStatisticsOffThenOnThenOff
com.hazelcast.cache.jsr.CacheManagerManagementTest.testCacheStatisticsOffThenOnThenOff
com.hazelcast.cache.jsr.CacheManagerManagementTest.testCacheManagementOffThenOnThenOff
com.hazelcast.cache.jsr.CacheManagerManagementTest.testCacheManagementOffThenOnThenOff
com.hazelcast.cache.jsr.CacheManagerManagementTest.testDoubleRegistration
com.hazelcast.cache.jsr.CacheManagerManagementTest.testDoubleRegistration
com.hazelcast.cache.jsr.CacheManagerManagementTest.testNoEntriesWhenNoCaches
com.hazelcast.cache.jsr.CacheManagerManagementTest.testNoEntriesWhenNoCaches
com.hazelcast.cache.jsr.CacheManagerManagementTest.testMultipleCacheManagers
com.hazelcast.cache.jsr.CacheManagerManagementTest.testMultipleCacheManagers
```

```
java.lang.AssertionError: 

Expected: an empty collection
     but: <[javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-34, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-79, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-71, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-23, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-60, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-3, javax.cache:type=CacheStatistics,CacheManager=,Cache=myCache, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-11, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-67, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-93, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-19, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-45, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-82, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-56, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-22, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-92, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-61, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-78, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-35, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-57, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-18, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-66, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-4, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-10, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-44, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-70, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-83, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-32, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-73, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-47, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-91, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-21, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-9, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-58, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-50, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-13, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-84, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-99, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-1, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-37, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-81, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-89, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-24, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-76, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-63, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-55, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-6, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-94, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-42, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-68, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-16, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-29, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-30, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-38, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-75, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-86, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-49, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-64, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-7, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-52, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-15, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-97, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-41, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-31, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-74, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-87, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-26, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-39, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-65, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-48, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-8, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-53, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-40, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-96, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-27, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-14, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-0, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-80, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-88, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-25, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-36, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-62, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-77, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-54, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-17, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-5, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-43, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-95, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-69, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-28, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-33, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-85, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-72, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-46, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-90, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-20, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-59, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-2, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-12, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-51, javax.cache:type=CacheStatistics,CacheManager=,Cache=cache-with-stats-98]>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.jsr107.tck.management.CacheManagerManagementTest.tearDown(CacheManagerManagementTest.java:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:402)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.AbstractHazelcastClassRunner$1.evaluate(AbstractHazelcastClassRunner.java:302)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:379)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:340)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:413)
```

--- a/hazelcast/src/test/java/com/hazelcast/internal/management/TimedMemberStateTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/management/TimedMemberStateTest.java
@@ -16,11 +16,11 @@
 
 package com.hazelcast.internal.management;
 
-import com.hazelcast.cache.CacheUtil;
 import com.hazelcast.cache.ICache;
 import com.hazelcast.cache.impl.CacheService;
 import com.hazelcast.config.CacheSimpleConfig;
 import com.hazelcast.config.Config;
+import com.hazelcast.core.DistributedObject;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.internal.json.JsonObject;
 import com.hazelcast.internal.monitor.impl.MemberStateImpl;
@@ -30,11 +30,15 @@ import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
+import java.util.Collection;
+
+import static com.hazelcast.cache.CacheUtil.getDistributedObjectName;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
@@ -63,6 +67,16 @@ public class TimedMemberStateTest extends HazelcastTestSupport {
         hz = createHazelcastInstance(config);
         timedMemberStateFactory = new TimedMemberStateFactory(getHazelcastInstanceImpl(hz));
         timedMemberState = createState();
+    }
+
+    @After
+    public void tearDown() {
+        // explicit cleanup is required because the MBean server is static so registrations
+        // will be left over when test's HazelcastInstance shuts down
+        Collection<DistributedObject> distributedObjects = hz.getDistributedObjects();
+        for (DistributedObject object : distributedObjects) {
+            object.destroy();
+        }
     }
 
     @Test
@@ -109,7 +123,7 @@ public class TimedMemberStateTest extends HazelcastTestSupport {
         hz.getCacheManager().getCache(CACHE_WITH_STATS_PREFIX + "1");
         CacheService cacheService = nodeEngine.getService(CacheService.SERVICE_NAME);
         assertNotNull(cacheService.getStats()
-                          .get(CacheUtil.getDistributedObjectName(CACHE_WITH_STATS_PREFIX + "1")));
+                          .get(getDistributedObjectName(CACHE_WITH_STATS_PREFIX + "1")));
     }
 
     @Test
@@ -127,8 +141,8 @@ public class TimedMemberStateTest extends HazelcastTestSupport {
 
         MemberStateImpl memberState = createState().getMemberState();
         for (int i = 0; i < 100; i++) {
-            assertNotNull(memberState.getLocalCacheStats(CacheUtil.getDistributedObjectName(CACHE_WITH_STATS_PREFIX + i)));
-            assertNull(memberState.getLocalCacheStats(CacheUtil.getDistributedObjectName(CACHE_WITHOUT_STATS_PREFIX + i)));
+            assertNotNull(memberState.getLocalCacheStats(getDistributedObjectName(CACHE_WITH_STATS_PREFIX + i)));
+            assertNull(memberState.getLocalCacheStats(getDistributedObjectName(CACHE_WITHOUT_STATS_PREFIX + i)));
         }
     }
 

